Use SessionAuthenticationStrategy for Remember-Me authentication

xhaggi opened the pull request:
As described in gh-2253, the `RememberMeAuthenticationFilter` does not call the configured `SessionAuthenticationStrategy` if there is one, and therefore concurrent session control does not work properly.

This PR adapts the `RememberMeAuthenticationFilter` so that it calls the `SessionAuthenticationStrategy` on successful authentication as the `AbstractAuthenticationProcessingFilter` does. The shared `SessionAuthenticationStrategy` is then used in the `RememberMeConfigurer` to configure the filter like it is done in `FormLoginConfigurer`.

Let me know if anything is missing!

<!--
For Security Vulnerabilities, please use https://pivotal.io/security#reporting
-->

<!--
Before creating new features, we recommend creating an issue to discuss the feature. This ensures that everyone is on the same page before extensive work is done.

Thanks for contributing to Spring Security. Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with gh-).
-->

jzheaux reviewed (changes requested):
Thanks for the PR, @xhaggi! Could you add a unit test that confirms the desired behavior (concurrency control) works, for example when the DSL configures a max number of sessions and the remember me filter exceeds that?

xhaggi commented:
@jzheaux of course. I have added the unit test.

xhaggi commented:
@jzheaux the [build](https://github.com/spring-projects/spring-security/actions/runs/11121314806/job/31517224220) failed because formatting violations were found in the files I changed. I fixed it and force-pushed the changes.

xhaggi commented:
BTW is there any chance that this will be backported to earlier versions?

jzheaux commented:
Hi, @xhaggi, no, it's not likely, given that there is a behavioral change and a new public API. That said, what version were you hoping to backport it to and what is preventing you from upgrading?

xhaggi commented:
>what version were you hoping to backport

I was thinking of the last stable version 6.3.x.

>what is preventing you from upgrading

Nothing, but we have to wait for the next stable 😉
